packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -114,6 +115,11 @@ export class ApolloServerBase {
   // the default version is specified in playground.ts
   protected playgroundOptions?: PlaygroundRenderPageOptions;
 
+  // An optionally defined store that, when enabled (default), will store the
+  // parsed and validated versions of operations in-memory, allowing subsequent
+  // parse and validates on the same operation to be executed immediately.
+  private documentStore?: InMemoryLRUCache<DocumentNode>;
```

Comment:
Should all the declarations (esp on GraphQLServerOptions) refer to the interface KeyValueCache instead of the implementation?

Reply:
I think we want to avoid supporting a more generic `KeyValueCache` than that implemented in `InMemoryLRUCache` since these are native objects and must be stored in _our_ memory.  I think it's worth also ensuring that extra functionality (e.g. `flush`) that might only be available on `InMemoryLRUCache` be exposed.  This would seem to still support keeping this typed as the `InMemoryLRUCache` implementation.